pkg/wal/processor/kafka/wal_kafka_batch_writer.go

```diff
@@ -138,10 +138,11 @@ func (w *BatchWriter) ProcessWALEvent(ctx context.Context, walEvent *wal.Event)
 		if err != nil {
 			return fmt.Errorf("marshalling event: %w", err)
 		}
-		// check if walEventBytes is larger than the Kafka accepted max message size
-		if len(walDataBytes) > int(w.maxBatchBytes) {
+		// check if walEventBytes is larger than 95% of the Kafka accepted max
+		// message size to allow for some buffer for the rest of the message
+		if len(walDataBytes) > int(0.95*float64(w.maxBatchBytes)) {
```

Comment:
this is probably good enough, although it can be a lot of buffer for high max bytes settings, I wonder if there is a predictable payload size math (ie 100bytes padding)

Reply:
I'll have a think/look.